contracts/src/TeleporterTokenSource.sol

```diff
@@ -143,9 +152,10 @@ abstract contract TeleporterTokenSource is ITeleporterTokenBridge, TeleporterOwn
                 "TeleporterTokenSource: invalid bridge address"
             );
             _withdraw(input.recipient, amount);
+            return;
```

Comment:
Not new in this PR, but I think the require in this if block is a bit confusing. 

IIUC, the bridge contracts support multiple destination contracts on the same blockchain, but it does not support bridging between two contracts on the same chain. This is a reasonable practical restriction, but I wonder if will just result in more errors (i.e. specifying the wrong bridge address for a given blockchain and then there is no recovery route). I think we should consider either:
- removing this restriction all together
- more clearly documenting
- requiring on `send` from a destination contract that if the `destinationBlockchainID` is the `sourceBlockchainID`, that the `destinationBridgeAddress` is the `sourceBridgeAddress`.

This is best for a separate PR, but wanted to call it out now.

Reply:
Didn't quite what what you meant here, this `if (input.destinationBlockchainID == blockchainID) {` block is only if the final destination is the "home chain" the source blockchain ID, so we withdraw. Wasn't sure if `blockchainID` variable naming was confusing.

We don't allow directly bridging between two destination bridge contracts, but we do allow a multihop between the two destinations on the same chain. Still good to have a check on destinations where if we're sending back to the `sourceBlockchainID` the destination bridge address also needs to be the right bridge address, so we fail before sending and don't get it stuck here.

Reply:
added better comment

Reply:
actually you're right there's an additional 
```
        require(
            input.destinationBlockchainID != blockchainID,
            "TeleporterTokenDestination: cannot bridge to same chain"
        );
```
that prevented bridging between two destination bridge instances on same chain, going to remove that

Reply:
I don't see the point in allowing bridging to the exact same destination bridge instance, so going to check against that.

Reply:
I don't see a point in allowing bridging to the exact same destination instance, so will check against that.